packages/loopring_v3/contracts/aux/access/IBlockReceiver.sol

```diff
@@ -11,10 +11,8 @@ import "../../amm/libamm/AmmData.sol";
 abstract contract IBlockReceiver
 {
     function beforeBlockSubmission(
-        ExchangeData.Block memory _block,
-        bytes              memory data,
-        uint                      txIdx,
-        uint                      numTxs
+        bytes              calldata txsData,
```

Comment:
Passing in a bytes instead of  ExchangeData.Block is cheaper? I thought they are all in memory and should be both very cheap.

Reply:
Not using `ABIEncoderV2` data structures is always cheaper no matter what. The simpler the data type the better it seems in all cases. So not using a `Block` datatype here (even though by encoding the auxiliary data as `bytes` makes is much simpler) is still a bit cheaper, and these can be called often (instead of a single call to submitBlocks where it doesn't matter much) so if not reallly needed best to avoid it.